source/MaterialXView/Viewer.cpp

```diff
@@ -1539,43 +1569,104 @@ void Viewer::loadShaderSource()
     }
 }
 
-void Viewer::saveDotFiles()
+void Viewer::saveDiagrams()
 {
```

Comment:
my understanding is that the non-generic part of the traversal is here, and in the test function elsewhere?